### ISSUE TITLE
torch.halt or torch.exit function implementation

### ISSUE THREAD

**Issue by bishshoy** [pytorch/pytorch#72099]
### 🚀 The feature, motivation and pitch

In multi-GPU training, sometimes I write an exit() function under an if statement. If the condition of the if statement gets satisfied by only one GPU, then the exit() function is called on that GPU and not on the rest of the GPUs. Is there any way I can indicate a halt or an exit() for all GPUs simultaneously? If not, is it possible to include such a function in the API itself?

Thanks.

**Comment by H-Huang** [comment]
Hi, could you answer the following questions about your setup?

- How are you performing multi-gpu training (are you using anything from torch.distributed package, e.g. DDP, RPC)?
- What condition are you calling exit on the GPU, is it at the end of training?
- Any minimal reproducible scripts that we can run locally would help in understanding the issue.

**Comment by bishshoy** [comment]
1) I am using DDP on a single node containing 4 GPUs.

2) Normally, I would call `exit()` somewhere in the middle of training (for debug purposes).

3)
```
# Compute loss
loss = nn.CrossEntropyLoss(output, target)
if torch.isnan(loss):
    exit(0)
```
When run on four GPUs, the code above only hits exit on the GPU that encountered the NaN loss. It would be great if we could _broadcast_ the exit call across all GPUs.

**Comment by pbelevich** [comment]
cc @kiukchung @aivanou

**Comment by bishshoy** [comment]
I have found a single line solution. It is to raise some kind of an exception. For example, if I issue the following exception on one of the GPUs:
```
# Compute loss
loss = nn.CrossEntropyLoss(output, target)
if torch.isnan(loss):
    raise ValueError('NaN encountered. Exiting...')
```
then torch sends a `SIGTERM` on all GPUs and kills all processes. This does the job for me. :)
